src/http/conversions.rs

```diff
@@ -526,30 +531,36 @@ impl TryIntoOutgoingRequest for OutgoingRequest {
 }
 
 impl TryIntoOutgoingRequest for Request {
-    type Error = std::convert::Infallible;
+    type Error = anyhow::Error;
 
     fn try_into_outgoing_request(self) -> Result<(OutgoingRequest, Option<Vec<u8>>), Self::Error> {
         let headers = self
             .headers()
             .map(|(k, v)| (k.to_owned(), v.as_bytes().to_owned()))
             .collect::<Vec<_>>();
-        let request = OutgoingRequest::new(
-            self.method(),
-            self.path_and_query(),
-            Some(if self.is_https() {
+        let request = OutgoingRequest::new(Headers::from_list(&headers)?);
+        request
+            .set_method(self.method())
+            .map_err(|()| anyhow::anyhow!("error setting method to {}", self.method()))?;
+        request
+            .set_path_with_query(self.path_and_query())
+            .map_err(|()| anyhow::anyhow!("error setting path to {:?}", self.path_and_query()))?;
+        request
+            .set_scheme(Some(if self.is_https() {
                 &super::Scheme::Https
             } else {
                 &super::Scheme::Http
-            }),
-            self.authority(),
-            &Headers::new(&headers),
-        );
+            }))
+            .unwrap();
```

Comment:
```suggestion
            ?;
```

Reply:
I chose `unwrap()` there because it can't error according to the `wasi-http` documentation.  It only errors on malformed `Scheme::Other`, but we never pass `Scheme::Other`.  I'll add a comment to that effect, though.